### DISCUSSION TITLE
how to make a dialog variation for a npc

### DISCUSSION THREAD

**Discussion by NickoBlackSaturn** [dialogic-godot/dialogic#1356] [Q&A]
Sorry if the question is silly, but I'm trying to figure out how to make npc "trigger" another timeline when I interact with him more than one time.

A little example

There are two timelines, one that contains the text "hello!" and the another has "How are you doing?"
When I talk to a npc he says "Hello", timeline 1.

But now, here I am. What I need to do to interact with this npc again and make him show the timeline that says "How are you doing?"

also sorry if my english is bad

**Comment by Jowan-Spooner** [comment]
Hi, this is a pretty common request. Usually you do this with a flag or counter variable. This could either be done in Dialogic or your script. Create a variable (e.g. Interaction_counter) . Each time the player interacts with the npc increase it by one. Then you can check and play a specific timeline depending on that counter.
Alternatively you can do it with dialogic variables and if events. Depending on your timelines this could make them quite messy though.

**Reply by NickoBlackSaturn** [reply]
It worked for me! I made a counter variable that increases +1 everytime the dialog unpauses the game.
So depending on how much this interaction counter has, it changes the timeline

One more doubt, doing this way is possible to make it choose a random timeline when I interact again?
